opencti-worker/src/worker.py

```diff
@@ -493,16 +490,13 @@ def run(self) -> None:
                         "Processing a new message, launching a thread...",
                         {"queue": self.queue_name, "tag": method.delivery_tag},
                     )
-                    task_future = self.execution_pool.submit(
+                    self.execution_pool.submit(
                         self.data_handler,
                         self.pika_connection,
                         self.channel,
                         method.delivery_tag,
                         data,
                     )
-                    while task_future.running():  # Loop while the thread is processing
-                        self.pika_connection.sleep(0.05)
-                    self.worker_logger.info("Message processed, thread terminated")
```

Comment:
I identified a potential concurrency issue that we had not initially considered. Indeed, this change allows multiple messages per queue to be processed concurrently. In the Consumer class, the OpenCTI API client (self.api) is modified at the beginning of data_handler (lines 320 to 329) and used later (lines 346, 372, 403, 424, and 447), which leaves room for a race condition between concurrent threads.
Note: In ApiConsumer, this appears to be safe.